### PR TITLE
fix(logging): delete logs when user identifier changes

### PR DESCRIPTION
**Pull request by phantumcode**
## Description
Update logger so that existing logs are deleted the user identifier changes due to sign in or sign out

## General Checklist

- [ ] ~Added new tests to cover change, if needed~
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [x] All integration tests pass
- [x] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] ~Documentation update for the change if required~
- [x] PR title conforms to conventional commit style
- [ ] ~New or updated tests include `Given When Then` inline code documentation and are named accordingly `testThing_condition_expectation()`~
- [ ] ~If breaking change, documentation/changelog update with migration instructions~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by thisisabhash** [comment]
Can we have a unit test for this change, if possible?

File: AmplifyPlugins/Logging/Tests/AWSCloudWatchLoggingPluginTests/AWSCloudWatchLoggingSessionControllerTests.swift [hunk 1]
```diff
@@ -67,6 +71,43 @@ final class AWSCloudWatchLoggingSessionControllerTests: XCTestCase {
...
     }
+    
+    /// Given: an AWSCloudWatchLoggingSessionController
+    /// When: the user identifier is changed
+    /// Then: a flush log is called and fails with a flushLogFailure hub event
+    func testResetLogsWhenIdentifierChanges() async throws {
+        let hubEventExpectation = expectation(description: "should receive flush failure event")
+        unsubscribeToken = Amplify.Hub.listen(to: .logging) { payload in
+            switch payload.eventName {
+            case HubPayload.EventName.Logging.flushLogFailure:
```
Comment (thisisabhash) [inline AmplifyPlugins/Logging/Tests/AWSCloudWatchLoggingPluginTests/AWSCloudWatchLoggingSessionControllerTests.swift:82]: is this expected? `flushLogs` failing when `userIdentifier` changes?

Comment (phantumcode) [reply AmplifyPlugins/Logging/Tests/AWSCloudWatchLoggingPluginTests/AWSCloudWatchLoggingSessionControllerTests.swift]: that's outdated, can you refresh and see latest updates?

File: AmplifyPlugins/Logging/Sources/AWSCloudWatchLoggingPlugin/Persistence/LogActor.swift [hunk 2]
```diff
@@ -56,4 +56,9 @@ actor LogActor {
             rotationSubject.send(log)
         }
     }
+    
+    func deleteLogs() throws {
+        try rotation.reset()
+        try rotation.currentLogFile.synchronize()
```
Comment (thisisabhash) [inline AmplifyPlugins/Logging/Sources/AWSCloudWatchLoggingPlugin/Persistence/LogActor.swift:62]: nit: we can call `self.synchronize()` here to maintain uniformity

**Review by thisisabhash — APPROVED** [review]
LGTM

**Review by thisisabhash — APPROVED** [review]

**Pull request merged**